Add Beam (#34)
makerdao/dss-flappers: README.md

* `maxRate` - Maximum allowed burn rate, measured as `kbump / hop` (the total surplus throughput). This caps the combined throughput even when `kbump` and `hop` are each individually within their own bound.
* `tau` - Cooldown period (in seconds) enforced between consecutive `set` calls.
* `toc` - Timestamp of the last `set` call.

`Splitter.burn` is bounded only at its upper end (`burn <= WAD`); it may be lowered freely down to zero.

Note that `maxKbump` must be safe as a standalone kick size, not just safe relative to `minHop` (max rate). Because `hop` only gates the gap since the last kick (`Splitter.zzz`), a long-idle or freshly-deployed Splitter lets a kick fire immediately regardless of `hop`.

Access control:
* `wards` (`rely`/`deny`) - Governance-level administrators that configure the ranges.
* `buds` (`kiss`/`diss`) - Facilitators permitted to call `set`.

> Halting: `set` is automatically blocked whenever the burn engine itself is stopped, i.e. when `Splitter.hop` is set to `type(uint256).max` or `Splitter.live` is set to `0`. This binds the `SBEBeam` halt state to the engine's real halt state — there is no separate flag to keep in sync — and it ensures a facilitator can never use `set` to revive a governance-halted engine. Only governance can bring it back, by re-filing a finite `hop` on the `Splitter`.

Reviewer: Would maybe remove (or reword) the last sentence since it doesn't apply to `Splitter.live`.

```suggestion
Halting: `set` is automatically blocked whenever the burn engine itself is stopped, i.e. when `Splitter.hop` is set to `type(uint256).max` or `Splitter.live` is set to `0`. This binds the `SBEBeam` halt state to the engine's real halt state — there is no separate flag to keep in sync — and it ensures a facilitator can never use `set` to revive a governance-halted engine.
```

> Halting: `set` is automatically blocked whenever the burn engine itself is stopped, i.e. when `Splitter.hop` is set to `type(uint256).max` (the canonical way governance disables kicks, since `Splitter.kick` then becomes unreachable). This binds the `SBEBeam` halt state to the engine's real halt state — there is no separate flag to keep in sync — and it ensures a facilitator can never use `set` to revive a governance-halted engine. Only governance can bring it back, by re-filing a finite `hop` on the `Splitter`.

Reviewer 2: ```suggestion
Halting: `set` is automatically blocked whenever the burn engine itself is stopped, i.e. when `Splitter.hop` is set to `type(uint256).max` or `Splitter.live` is set to `0`. This binds the `SBEBeam` halt state to the engine's real halt state — there is no separate flag to keep in sync — and it ensures a facilitator can never use `set` to revive a governance-halted engine. Only governance can bring it back, by re-filing a finite `hop` on the `Splitter`.
```

This contract allows bypassing the governance delay when disabling the Splitter in an emergency.

> ### FarmOwner

Reviewer 2: A bit too much documentation to my personal taste.
For example not sure we need to detail all the forwarded messages, or we need the `Access control` part.

Author: Removed the forwarded messages section
